peps/pep-0729.rst

```diff
@@ -0,0 +1,414 @@
+This is similar in spirit to the reasons why the Steering Council sometimes uses
+PEP delegation.
+
+Specification
+=============
+
+We propose the creation of a new group, the Typing Council. This group will
+be responsible for developing and maintaining the Python type system, and
+for solving the above problems.
+
+The "operations and process" section describes how this group would operate and
+be governed.
+
+The more exciting "projects" section describes solutions to the above problems
+that the Typing Council could shepherd.
+
+Mandate
+-------
+
+The Typing Council's mandate is to ensure that the Python type system is:
+
+* **Useful**: The type system should serve common use cases. As identified
+  by :pep:`484`, the primary use case is static analysis, but there are others,
+  such as runtime type checking, static compilation, IDE support, and documentation.
+  The Typing Council should consider all of these use cases when making decisions,
+  and be open to supporting additional use cases as they come up.
+* **Usable**: The type system should be easy to use for Python developers. It
+  should be ergonomic to write well-typed Python code that is accepted by type
+  checkers. There should be good documentation for the type system.
+* **Stable**: As the type system matures, users should be able to rely on their
+  typed code continuing to work and be able to trust their mental model for the
+  type system. Changes should be made with care and in a way
+  that minimizes disruption. Nevertheless, the type system should be able to
+  evolve, and type checkers need not be held to the same strict compatibility
+  standards as the Python language itself.
+
+Operations and process
+----------------------
+
+The council would have three members, comprised of prominent community members,
+such as Python core developers and maintainers of major type checkers. The members
+should include people affiliated with a variety of projects related to type checking,
+which may include type checkers, CPython, typeshed, or other projects.
+
+The Steering Council would appoint the initial Typing Council. There is no term
+limit for council members. Council members may resign their position at any time.
+There is a norm that at least one person will voluntarily resign from the
+Typing Council each year.
+
+To determine replacements, nominations will be collected from the typing
+community. Self-nominations are allowed. The existing Typing Council will then decide
+the replacement member(s) from the nominees. The expectation is that this would
+be done by fiat, but the Typing Council can choose a replacement by any means
+they see fit, including a vote.
+
+The Typing Council remains accountable to the Steering Council. At any point,
+for any reason, the Steering Council could (publicly or privately) make a
+specific change or request a non-specific change to the composition of the
+Typing Council.
```

Comment:
The SC ultimately has authority over the type system. But I don't think the SC should make changes directly... If SC ever wanted to override TC, I feel that should in the form of a PEP that the SC accepts. Similarly, if the event of irreconcilable TC deadlock about type system, it sounds right that there should be a PEP that the SC decides on.

(Note in this version of the draft that all PEPs go to the SC by default)

Reply:
Okay, agreed they should still go through the PEP process. It's interesting though, since the process proposed in this PEP explicitly allows some "minor" changes to the spec to be made without a PEP. I'm guessing that once there's a serious disagreement, by definition it's no longer "minor".